stl/inc/xutility

```diff
@@ -3662,27 +3674,40 @@ _DestTy* copy(_ExPo&&, _FwdIt1 _First, _FwdIt1 _Last, _DestTy (&_Dest)[_DestSize
 #endif // _HAS_CXX17
 
 // FUNCTION TEMPLATE copy_n
+template <class _InIt, class _Diff, class _OutIt>
+constexpr _OutIt _Copy_n_core(_InIt _First, _Diff _Count, _OutIt _Dest) {
+    for (;;) {
+        *_Dest = *_First;
+        ++_Dest;
+        --_Count;
+        if (_Count == 0) { // note that we avoid an extra ++_First here to allow istream_iterator to work,
+                           // see LWG-2471

```

Comment:
**LWG-2471 RAGE INTENSIFIES**

I can't get over how broken this is: `copy_n(i, 0, r)` needs to force `*i` back into the stream to be consistent with `copy_n(i, 1, r)`. 

LWG-2242 and LWG-2471 are the "`std::copy_n` is a horrible mess that we can't fix without breaking existing code" issues.

(No change requested.)

Reply:
I still don't understand why you want the algorithm to do an extra increment it doesn't otherwise have to do, particularly when doing the one less increment makes some use cases work.

(But this is also off topic for this review :) )